keyboards/Emi20/rules.mk

```diff
@@ -0,0 +1,57 @@
+# MCU name
+MCU = atmega32u4
+
+# Processor frequency.
+#     This will define a symbol, F_CPU, in all source code files equal to the
+#     processor frequency in Hz. You can then use this symbol in your source code to
+#     calculate timings. Do NOT tack on a 'UL' at the end, this will be done
+#     automatically to create a 32-bit value in your source code.
+#
+#     This will be an integer division of F_USB below, as it is sourced by
+#     F_USB after it has run through any CPU prescalers. Note that this value
+#     does not *change* the processor frequency - it should merely be updated to
+#     reflect the processor speed set externally so that the code can use accurate
+#     software delays.
+F_CPU = 16000000
+
+#
+# LUFA specific
+#
+# Target architecture (see library "Board Types" documentation).
+ARCH = AVR8
+
+# Input clock frequency.
+#     This will define a symbol, F_USB, in all source code files equal to the
+#     input clock frequency (before any prescaling is performed) in Hz. This value may
+#     differ from F_CPU if prescaling is used on the latter, and is required as the
+#     raw input clock is fed directly to the PLL sections of the AVR for high speed
+#     clock generation for the USB and other AVR subsections. Do NOT tack on a 'UL'
+#     at the end, this will be done automatically to create a 32-bit value in your
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE ?= no	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
+COMBO_ENABLE = yes
```

Comment:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
COMBO_ENABLE = yes
```

Reply:
The Combo feature cannot be enabled at the keyboard level. It will cause the Configurator to fail when trying to compile because it requires custom code in the keymap that it does not generate.

Reply:
Wait, so it shouldn't compile? I tried to compile this before making the pull request and it did compile. So what should I do instead? or just remove the COMBO_ENABLE = yes? as I do have custom keycodes in the keymap.c file?

Reply:
It will compile locally, but when the PR is merged it won't compile in the Configurator. You need to remove it from here and add a keymap-level rules.mk instead with just:
```makefile
COMBO_ENABLE = yes
```

Reply:
Would this still compile locally? I added a rules.mk file with just:

`COMBO_ENABLE = yes`

In the folder where the keymap.c file is also located, this did not compile locally.

Reply:
Ah, you also need to move `COMBO_COUNT` into a keymap-level config.h:
```c
#pragma once

#define COMBO_COUNT 1
```

Reply:
:/ That still doesn't do it, I did use `qmk clean` to make sure nothing there was interfearing. I just get errors that a combo isn't set yet so it doesn't recognise the lines with combo... in it. I made the files:

`config.h`
```
#pragma once

#define COMBO_COUNT 1
```
and
`rules.mk`
```
COMBO_ENABLE = yes
```

Should I make a new pull request so you can check out the files or upload them here in the comments? Or is this more of a issue to be discust in the discord server?

![image](https://user-images.githubusercontent.com/56627792/126544762-04a06b6d-43bf-49f3-9baa-17cffd02adbb.png)

Reply:
Just push the changes you've got so far to this branch and we can go from there.

Reply:
I tried to push all the changes, doesn't look like it changed everything tho :/ I'll have another look maybe I didn't fully add the changes to the linked qmk folder

Reply:
I recommend you just apply the suggestions, then pull them down locally before applying any other changes.

Reply:
I did do that, but on my my drive here. But I guess you mean commiting the changes you suggested?

Reply:
I mean applying the suggestions made here on GitHub.